File: Version2/JAVA/ReadCurrenciesRestApi.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you: hire me!
 */
package JAVA;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

public class ReadCurrenciesRestApi {

    public static void main(String[] args) throws IOException {
        String site = "https://kriptonita.com.br";

        String[] ar = getCurrenciesWithREST( site );
        for (String moeda : ar) {
            System.out.println( "Moeda: " + moeda ); 
        }
	}

    /**
     * Simple example of REST API call to return all available currencies of Magento store.
     */
    public static String[] getCurrenciesWithREST(String site) {
        List<String> results = new ArrayList<String>();

        try {
            URL url = new URL(site + "/rest/V1/directory/currency");
            HttpURLConnection response = (HttpURLConnection) url.openConnection();

            if ( response.getResponseCode() == 200 ) {
                BufferedReader contents = new BufferedReader(new InputStreamReader(response.getInputStream()));
                String line = contents.readLine();

                if ( line.contains( "available_currency_codes" ) ) {
                    String[] currencies = line.substring( line.indexOf("available_currency_codes\":[") + 27, line.indexOf("]") ).split(",");

                    for (String currency : currencies) {
                        results.add( currency.replace('"', ' ').trim() );
                    }
                }
            }
            response.disconnect();

        } catch (IOException e) {
            e.printStackTrace();
        }

        return results.toArray(new String[results.size()]);
    }
}

File: Version2/JS/ReadCurrenciesRestApi.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you: hire me!
 */

//getCurrenciesWithREST('https://kriptonita.com.br');

/**
 * Simple example of REST API call to return all available currencies of Magento store.
 */
function getCurrenciesWithREST(site = "") {
    let result = [];

    const https = require('https');
    const url = `${site}/rest/V1/directory/currency`;
    
    https.get(url, (resp) => {
        let data = '';
        resp.on('data', (chunk) => {
            data += chunk;
        });

        resp.on('end', () => {
            data = JSON.parse(data);

            result = data["available_currency_codes"];
        })
    }).on('error', (err) => {
        console.log('Error: ' + err.message);
    });

    return result;
}
module.exports = {getCurrenciesWithREST}
